cityofaustin/atd-moped
# Make school zone beacons from open dataset available to choose in components form

## Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/18533

## Testing
**URL to test:** 

Locally

**Steps to test:**

Add a new component to the map,  component type Signal - School Zone Beacon. There will be an autocomplete that appears once youve chosen that component type. 

![Screenshot 2024-09-17 at 10 47 37 AM](https://github.com/user-attachments/assets/00b4602a-9ada-442f-b2db-5bde705a921f)

you can search for a specific beacon, or scroll to find one. Note that the knack/socrata dataset inserts extra spaces between the words, so if you are looking for `1406 Dale` you will need to search `1406  Dale`

Save the component. Much like when adding a signal, the create component flow bypasses the manually mapping step and zooms the map to what has been added. 

Test editing an existing Beacon, confirm the correct beacon appears in the autocomplete. Only change the attributes. Save. 
Test editing an existing Beacon by switching it to a Signal. Confirm the old beacon does not appear on the map and is deleted when checking the table
```
SELECT * FROM feature_school_beacons fsb 
```
Test switching a signal to a beacon.

Test deleting the component. The beacon is no longer associated with the project. 

---
#### Ship list
